thunder/tests/opinfos.py

```diff
@@ -2531,6 +2531,67 @@ def where_error_generator(op, device, dtype=torch.float32, **kwargs):
 conditional_and_mask_ops.append(where_opinfo)
 
 
+def nan_to_num_sample_generator(op, device, dtype, requires_grad, **kwargs):
+    make = partial(make_tensor, device=device, dtype=dtype, requires_grad=requires_grad)
+
+    a = make((4, 4), dtype=dtype, requires_grad=requires_grad)
+    double_max = torch.finfo(torch.float64).max
+    double_min = torch.finfo(torch.float64).max
+    if dtype == torch.FloatType:
+        a = torch.tensor((0, float("nan"), float("inf"), -float("inf")))
+    elif dtype == torch.ComplexType:
+        a = torch.tensor((complex(0, 0), complex(float("nan"), float("nan")), complex(float("inf"), -float("inf"))))
```

Comment:
nit: this code would probably be more readable and easier to write by assigning to the names `nan` `inf` and `neginf` like `nan = float('nan')`

Let's also add the following numbers to the complex tensor:

- `complex(nan, 0)`
- `complex(nan, inf)`
- `complex(inf, 0)`
-  `complex(0, inf)`
- `complex(-inf, 0)` 
- `complex(nan, 5)`
- `complex(inf, 3)`